docs/build.gradle

```diff
@@ -16,10 +16,29 @@ apply plugin: 'org.antora'
 
 description = 'Micrometer Docs'
 
+// Spring 6 requires Java 17+
+// skip this module when building with jdk <17
+if (!javaLanguageVersion.canCompileOrRun(17)) {
+    project.tasks.configureEach { task -> task.enabled = false }
+}
+
 repositories {
     mavenCentral()
 }
 
+java {
+    toolchain {
+        languageVersion = javaLanguageVersion
+    }
+}
+
+compileTestJava {
+    // need to override the config applied in the root build.gradle to all subprojects
+    // TODO can we not do this with Gradle's toolchains instead?
```

Comment:
My hope was the above `java` block would take care of setting the Java target/source for the test compile task as well, but it didn't. I'm not sure if there's a better way to do it than I did here, but I ran out of patience looking into ways to configure Gradle.